features/cellular/framework/AT/AT_CellularNetwork.cpp

```diff
@@ -197,6 +197,10 @@ void AT_CellularNetwork::urc_cgreg()
 nsapi_error_t AT_CellularNetwork::set_credentials(const char *apn,
                                                   const char *username, const char *password)
 {
+    if (!is_supported(AT_CGAUTH)) {
```

Comment:
Should there be check that if onny apn is set then it does not matter if authentication is supported?

Reply:
Good point due to APN can be given without username/password. See fixed.